core/drand_control.go

```diff
@@ -176,6 +183,13 @@ func (d *Drand) runDKG(leader bool, group *key.Group, timeout uint32, randomness
 	return finalGroup, nil
 }
 
+func (d *Drand) cleanupDKG() {
```

Comment:
You can require the lock in this funciton since currently you do require it before on both calls

Reply:
all of the calls do other things while holding that lock though, so i think it's better for the lock to be obtained by the caller as it is now